### PR TITLE
Migrate microbrenchmarks to null safety

### PR DESCRIPTION
**Pull request by Abhishek01039**
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Abhishek01039** [comment]
Hello @jmagman 
can you please review this PR?
Thanks

File: dev/benchmarks/microbenchmarks/lib/language/sync_star_semantics_bench.dart [hunk 1]
```diff
@@ -93,7 +93,7 @@ Iterable<InlineSpanSemanticsInformation> combineSemanticsInfoSyncStar(List<Inlin
       workingText += info.text;
       workingLabel ??= '';
       if (info.semanticsLabel != null) {
-        workingLabel += info.semanticsLabel;
+        workingLabel += info.semanticsLabel!;
```
Comment (jmagman) [inline dev/benchmarks/microbenchmarks/lib/language/sync_star_semantics_bench.dart:96]: Put it in a local, then you don't need the `!`.
```dart
      final String infoSemanticsLabel = info.semanticsLabel;
      if (infoSemanticsLabel != null) {
        workingLabel += infoSemanticsLabel;
```

File: dev/benchmarks/microbenchmarks/lib/language/sync_star_semantics_bench.dart [hunk 2]
```diff
@@ -122,7 +122,7 @@ Iterable<InlineSpanSemanticsInformation> combineSemanticsInfoList(List<InlineSpa
...
```
Comment (jmagman) [inline dev/benchmarks/microbenchmarks/lib/language/sync_star_semantics_bench.dart:125]: Same here.

**Review by jmagman — CHANGES_REQUESTED** [review]
microbenchmarks_ios passes on this PR.  Looks good with nits.

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**